modules/router-store/src/router_store_module.ts

```diff
@@ -33,11 +33,11 @@ export type RouterNavigationPayload<T> = {
 /**
  * An action dispatched when the router navigates.
  */
-export type RouterNavigationAction<T = RouterStateSnapshot> = {
-  type: typeof ROUTER_NAVIGATION;
-  payload: RouterNavigationPayload<T>;
-};
 
+export class RouterNavigationAction<T = RouterStateSnapshot> implements Action {
+  readonly type = ROUTER_NAVIGATION;
+  constructor(public payload: RouterNavigationPayload<T>) {}
+}
```

Comment:
after typescript 2.8 landed which support conditional type, I think ngrx team will stick with interface based Action

Reply:
Oh I did not know about this!
But since we're still in 2.7 this is the only way to do it.

Reply:
If we need to use classes here, make them abstract